corehq/form_processor/utils/general.py

```diff
@@ -26,7 +26,18 @@ def clear_local_domain_sql_backend_override(domain):
     _thread_local.use_sql_backend = use_sql_backend_dict
 
 
+def _short_circuit_sql_check(domain_object_or_name):
+    from corehq.apps.domain.models import Domain
+    domain_name = domain_object_or_name
+    if isinstance(domain_object_or_name, Domain):
+        domain_name = domain_object_or_name.name
+
+    return ENTERPRISE_OPTIMIZATIONS.enabled(domain_name)
```

Comment:
Unfortunately this can't be set for eNikshay since it's still running on a cluster with non-SQL domains.

To avoid confusion I've completely removed the toggle and just use the global setting